src/main/resources/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -0,0 +1,278 @@
+    StepResponse:
+      title: StepResponse
+      type: object
+      description: A Step within a Goal that must be completed in order for the overall Goal to be completed.
+      properties:
+        stepReference:
+          type: string
+          description: A unique reference for the Step
+          example: d38a6c41-13d1-1d05-13c2-24619966119b
+        title:
+          type: string
+          description: A title describing the step
+          example: Book communication skills course
+        targetDate:
+          type: string
+          format: date
+          description: A date by which the prisoner should aim to complete the step.
+          example: '2023-08-19'
+        status:
+          $ref: '#/components/schemas/StepStatus'
+        sequenceNumber:
+          type: integer
+          description: The number (position) of the Step within the overall Goal.
+          example: 1
+
+    CreateGoalRequest:
+      title: Goal
+      type: object
+      description: A request to create an individual Goal that a Prisoner aims to complete as part of their Action Plan. The goal's status will be set to 'ACTIVE'.
+      properties:
+        title:
+          type: string
+          description: A title explaining the aim of the goal.
+          example: Improve communication skills
+        reviewDate:
+          type: string
+          format: date
+          description: An ISO-8601 date representing when the Goal is up for review. Defaults to 6 months.
+          example: '2023-12-19'
+        steps:
+          type: array
+          description: A List of at least one Step.
+          items:
+            $ref: '#/components/schemas/CreateStepRequest'
+        notes:
+          type: string
+          description: Some additional notes related to the Goal.
+          example: Pay close attention to Peter's behaviour.
+      required:
+        - title
+        - reviewDate
+    CreateStepRequest:
+      title: CreateStepRequest
+      type: object
+      description: A request to create a Step within a Goal. The Step is created a status of NOT_STARTED.
+      properties:
+        title:
+          type: string
+          description: A title describing the step
+          example: Book first aid course
+        targetDate:
+          type: string
+          format: date
+          description: A date by which the prisoner should aim to complete the step.
+          example: '2023-12-19'
+        sequenceNumber:
+          type: integer
+          description: The number (position) of the Step within the overall Goal.
+          example: 1
+
+    GoalStatus:
+      title: GoalStatus
+      description: Enum describing the various statuses of a Goal's lifecyle.
+      type: string
+      enum:
+        - ACTIVE
+        - COMPLETE
+        - ARCHIVED
+    StepStatus:
+      title: StepStatus
+      description: Enum describing the various statuses of a Step's lifecyle.
+      type: string
+      enum:
+        - NOT_STARTED
```

Comment:
👍 for using uppercase enums in the REST API 👍 
(Unlike our previous project where they standardised on lowercase kebab case, which resulted in generated model enums having the word `MINUS` in them !!! - eg: `not-started` would be generated as `NOT_MINUS_STARTED` !!!)